***************************************************************************************
This patch was taken from the kernel-image-2.4.17-ia64-011226.13 Debian source package,
where it was named patch.guid-byte-order
***************************************************************************************
From: Richard Hirst <rhirst@linuxcare.com>
Subject: likely fix to /proc/efi/vars filenames
To: bdale@hp.com
Date: Tue, 2 Apr 2002 15:31:00 +0100

Hi Bdale,
  I think this is what is needed, on top of yesterdays patch.  Testing
here now; I've poked Matt about it and will let you know what he says.

Richard


--- linux/include/asm-ia64/efi.h-	Tue Apr  2 15:39:00 2002
+++ linux/include/asm-ia64/efi.h	Tue Apr  2 15:39:41 2002
@@ -248,8 +248,8 @@
 efi_guid_unparse(efi_guid_t *guid, char *out)
 {
 	sprintf(out, "%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x",
-		guid->b[0], guid->b[1], guid->b[2], guid->b[3],
-		guid->b[4], guid->b[5], guid->b[6], guid->b[7],
+		guid->b[3], guid->b[2], guid->b[1], guid->b[0],
+		guid->b[5], guid->b[4], guid->b[7], guid->b[6],
 		guid->b[8], guid->b[9], guid->b[10], guid->b[11],
 		guid->b[12], guid->b[13], guid->b[14], guid->b[15]);
         return out;



